File: fmarket/src/main/java/ro/fmarket/model/subscriber/Subscriber.java
package ro.fmarket.model.subscriber;

import javax.persistence.Column;
import javax.persistence.Entity;

import org.hibernate.annotations.Type;
import org.joda.time.DateTime;

import lombok.Data;
import ro.fmarket.core.base.BaseEntity;

@Data
@Entity
public class Subscriber extends BaseEntity {

	@Column(nullable = false, unique = true, length = 60)
	private String email;

	@Type(type = "org.jadira.usertype.dateandtime.joda.PersistentDateTime")
	private DateTime subscribeDate;

	@Type(type = "org.jadira.usertype.dateandtime.joda.PersistentDateTime")
	private DateTime unsubscribeDate;

	@Column(nullable = false, length = 50, unique = true)
	private String unsubscribeToken;

}

File: fmarket/src/main/webapp/app/models/subscriber.ts
export class Subscriber{
	public id:string;
	public description: string;
	public email: string;
	public subscribeDate: Date;
	public unsubscribeDate: Date;
	public unsubscribeToken: number;

	constructor(id: string, description: string, email: string, subscribeDate:Date, unsubscribeDate: Date, unsubscribeToken: number ) {
		this.id = id? id : "";
		this.description = description? description : "";
		this.email = email? email : "";
		this.subscribeDate = subscribeDate? subscribeDate.toLocaleDateString("en-US") : new Date(1,1,1,0,0,0,0).toLocaleDateString("en-US");
		this.unsubscribeDate = unsubscribeDate? unsubscribeDate.toLocaleDateString("en-US") : new Date(1,1,1,0,0,0,0).toLocaleDateString("en-US");
		this.unsubscribeToken = unsubscribeToken? unsubscribeToken : -1;
	}
}